libsoftwarecontainer/src/softwarecontainer.cpp

```diff
@@ -165,45 +165,91 @@ void SoftwareContainer::addGateway(Gateway *gateway)
     m_gateways.push_back(std::move(std::unique_ptr<Gateway>(gateway)));
 }
 
-void SoftwareContainer::setGatewayConfigs(const GatewayConfiguration &gwConfig)
+ReturnCode SoftwareContainer::setGatewayConfigs(const GatewayConfiguration &gwConfig)
 {
-    /* TODO:
-     * The below assumption is incorrect, we do not want to appy configs to
-     * activated gateways.
-     */
-    // Go through the active gateways and check if there is a configuration for it
-    // If there is, apply it.
+    ReturnCode result = ReturnCode::SUCCESS;
 
+    result = configureGateways(gwConfig);
+    if (isError(result)) {
+        return result;
```

Comment:
Yes, we do. Let's fix this when the pending discussion about this has resulted in some conclusion.